packages/typespec-test/test/multiclient_modular/generated/typespec-ts/src/api/foo/operations.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { RequestOptions } from "../../common/interfaces.js";
+import { isUnexpected, Client } from "../../rest/foo/index.js";
+import { Resource, CustomPage } from "./models.js";
+
+export interface CreateOrUpdateOptions extends RequestOptions {
+  /** */
+  description?: string;
+  /** Accept header. */
+  accept?: "application/json";
+  /** Body parameter Content-Type. Known values are: application/json. */
+  content_type?: string;
+}
+
+/** Creates a new resource or updates an existing one. */
+export async function createOrUpdate(
+  context: Client.FooContext,
+  name: string,
+  type: string,
+  name: string,
+  options: CreateOrUpdateOptions = { requestOptions: {} }
+): Promise<Resource> {
+  const result = await context.path("/cadl-foo/resources/{name}", name).put({
```

Comment:
In this case, the body parameter is a Resource type which has two required properties name and type, however name is already a path level parameter, I wonder if purely exlude the path parameter from the required body flatten parameter is enough ？ 
Here's the typespec definition   
```
  @doc("Creates a new resource or updates an existing one.")
  createOrUpdate is ResourceCreateOrReplace<Resource>;
```

Reply:
turns out I should set the visibility of name as read because of key.